src/mmrelay/matrix_utils.py

```
@@ -2955,125 +3179,180 @@ async def on_room_message(
 
-    return response
+    try:
+        response, _ = await client.upload(
+            io.BytesIO(image_data),
+            content_type=content_type,
+            filename=filename,
+            filesize=len(image_data),
+        )
+    except NIO_COMM_EXCEPTIONS as e:
+        # Convert nio communication exceptions to an UploadError instance
+        logger.exception(f"Image upload failed due to a network error: {e}")
```

Ruff (0.14.5)

Redundant exception object included in `logging.exception` call

(TRY401)